### PR TITLE
test: Add e2e test to access instance using DNS TXT record. part of #842

### PR DESCRIPTION
**Pull request by hessjcg**
Add e2e test to verify that connecting with a DNS name works correctly.

### CODE REVIEW

File: e2e_mysql_test.go [hunk 1]
```diff
@@ -54,25 +57,52 @@ func requireMySQLVars(t *testing.T) {
 	}
 }
 
+type mockResolver struct {
+}
+
+func (r *mockResolver) Resolve(_ context.Context, name string) (instanceName instance.ConnName, err error) {
+	if name == "mysql.example.com" {
+		return instance.ParseConnNameWithDomainName(mysqlConnName, "mysql.example.com")
```
Comment (jackwotherspoon) [inline e2e_mysql_test.go:65]: is this what we would expect a global DNS format to look like? Or would it be in the form with the trailing dot similar to PSC?

Comment (hessjcg) [reply e2e_mysql_test.go:65]: The domain name is customer-defined and customer-entered, so I'd expect it to omit the trailing dot.

File: e2e_mysql_test.go [hunk 2]
```diff
@@ -85,18 +115,18 @@ func TestMySQLDriver(t *testing.T) {
...
 			if err != nil {
 				t.Fatalf("failed to register driver: %v", err)
 			}
 			defer cleanup()
 			cfg := gomysql.NewConfig()
 			cfg.CheckConnLiveness = true
-			cfg.User = mysqlUser
-			cfg.Passwd = mysqlPass
+			cfg.User = tc.user
+			cfg.Passwd = tc.password
```
Comment (jackwotherspoon) [inline e2e_mysql_test.go:126]: why do we need to be changing the user and pass per test case?

Comment (hessjcg) [reply e2e_mysql_test.go:126]: The test should use the user/password mysqlUser and mysqlPassword for normal connections. The test should use mysqlIAMUser and "password" when the test includes the option WithIAMAuthN().

Comment (jackwotherspoon) [reply e2e_mysql_test.go:126]: It looks like the MySQL IAM authN test wasn't even doing and IAM AuthN login previously then... as `mysqlIAMUser` was unused from what I can see?

**Review by jackwotherspoon — APPROVED** [review]
LGTM 👍

**Pull request merged**